src/psyclone/psyir/nodes/nan_test_node.py

```diff
@@ -0,0 +1,147 @@
+    :type children: list of :py:class:`psyclone.psyir.nodes.Node`
+    :param parent: the parent of this node in the PSyIR tree.
+    :type parent: :py:class:`psyclone.psyir.nodes.Node`
+    :param options: a dictionary with options provided via transformations.
+    :type options: dictionary of string:values or None
+    :param str options["prefix"]: a prefix to use for the PSyData module name \
+        (``prefix_psy_data_mod``) and the PSyDataType
+        (``prefix_PSyDataType``) - a "_" will be added automatically. \
+        It defaults to "extract", which means the module name used will be \
+        ``extract_psy_data_mode``, and the data type ``extract_PSyDataType``.
+
+    '''
+    # Textual description of the node.
+    _text_name = "NanTest"
+    _colour_key = "NanTest"
+
+    def __init__(self, ast=None, children=None, parent=None, options=None):
+        if options:
+            my_options = options.copy()
+        else:
+            my_options = {}
+        # If there is no value specified by in the constructor, default
+        # to the "extract" class.
+        my_options["prefix"] = my_options.get("prefix", "nan_test")
+        super(NanTestNode, self).__init__(ast=ast, children=children,
+                                          parent=parent, options=my_options)
+
+
+    @property
+    def nan_test_body(self):
+        '''
+        :returns: the Schedule associated with this NanTestNode.
+        :rtype: :py:class:`psyclone.psyir.nodes.Schedule`
+
+        '''
+        return super(NanTestNode, self).psy_data_body
+
+    @property
+    def dag_name(self):
+        '''
+        Returns the name to use in a DAG for this Node
+
+        :returns: the dag name of ExtractNode.
+        :rtype: str
+        '''
+        return "nan_test_" + str(self.position)
+
+    def update_vars_and_postname(self):
+        '''
+        This function is called after the variables to be extracted
+        have been stored in self._input_list and self._output_list.
+        It can be used to e.g. remove unnecessary variables (e.g. loop
+        counter), or adjust the postfix to assure that no duplicated
+        variable name is created. This default function does not
+        do anything atm.
+        '''
+
+    def gen_code(self, parent):
+        # pylint: disable=arguments-differ
+        '''
+        Generates the code required for verification of the parameters of
+        one or more Nodes. It uses the PSyData API (via the base class
+        PSyDataNode) to create the required callbacks that will allow a
+        library to write the kernel data to a file.
+
+        :param parent: the parent of this Node in the PSyIR.
+        :type parent: :py:class:`psyclone.psyir.nodes.Node`.
+        '''
+
+        # Determine the variables to write:
+        from psyclone.psyir.tools.dependency_tools import DependencyTools
+        dep = DependencyTools()
+        self._input_list, self._output_list = dep.get_in_out_parameters(self)
```

Comment:
pylint complains that these two private attributes are defined outside the constructor.

Reply:
Fair enough. I don't need the whole callback sequence here, so I converted this all into normal variables (fancy for: copy&paste error)